pkg/workers/worker.go

```diff
@@ -0,0 +1,32 @@
+package workers
+
+import (
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/utils"
+)
+
+type WorkHandler[Req, Res any] func(Req) Res
+
+func GoWorker[Req, Res any](
+	reqChan chan Req,
+	resChan chan utils.Pair[Req, Res],
```

Comment:
- just a thought, will it make more sense to also specify the intention of these channels not only in their names, like
```suggestion
	reqChan <- chan Req,
	resChan chan <- utils.Pair[Req, Res],
```

Reply:
This is new to me. Is this a way to force Go to only allow sends and not receives?

Reply:
- `req <-chan`: forces `req` to be used only for `receive` in function body
- `res chan<-`: forces `res` to be used only for `send` in function body